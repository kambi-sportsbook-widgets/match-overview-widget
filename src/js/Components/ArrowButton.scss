:local(.left),
:local(.right) {
   position: absolute;
   top: -12px;
   bottom: -9px;
   width: 7%;
   min-width: 46px;
   border: none;
   background: none;
   font-size: 15px;
   font-weight: bold;
   cursor: pointer;
   outline: none;
   padding: 0;
   transition: opacity .3s;

   &[disabled] {
      opacity: 0;
      cursor: default;
   }

   :local(.icon) {
      fill: #fff;
      opacity: .15;
      transition: opacity .3s;
      margin: 0 12px;
   }

   &:hover :local(.icon) {
      opacity: 1;
   }
}

:local(.left) {
   background: linear-gradient(90deg, #000 0%, transparent);
   left: 0;
   text-align: left;
}

:local(.right) {
   background: linear-gradient(270deg, #000 0%, transparent);
   right: 0;
   text-align: right;

   :local(.icon) {
      transform: scale(-1, 1);
   }
}
