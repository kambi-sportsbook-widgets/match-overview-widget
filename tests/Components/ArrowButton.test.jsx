/* eslint-env jest */
import React, { Children } from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { shallow } from 'enzyme';
import ArrowButton from '../../src/js/Components/ArrowButton';

let renderer;

describe('ArrowButton DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });

   it('renders correctly left variant', () => {
      expect(renderer.render(
         <ArrowButton type="left" onClick={() => {}} />
      )).toMatchSnapshot();
   });

   it('renders correctly right variant', () => {
      expect(renderer.render(
         <ArrowButton type="right" onClick={() => {}} />
      )).toMatchSnapshot();
   });

   it('renders correctly when disabled', () => {
      expect(renderer.render(
         <ArrowButton type="right" onClick={() => {}} disabled={true} />
      )).toMatchSnapshot();
   });

});

describe('ArrowButton behaviour', () => {

   it('handles clicks correctly', () => {
      const onClickMock = jest.fn();

      const wrapper = shallow(
         <ArrowButton type="left" onClick={onClickMock} />
      );

      expect(onClickMock).not.toHaveBeenCalled();

      wrapper.find('button').simulate('click');

      expect(onClickMock).toHaveBeenCalledTimes(1);
   });

});
