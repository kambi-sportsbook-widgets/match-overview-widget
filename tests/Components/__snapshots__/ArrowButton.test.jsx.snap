// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowButton DOM rendering renders correctly left variant 1`] = `
<button
  className="left"
  disabled={false}
  onClick={[Function]}
>
  <svg
    className="icon"
    height="36px"
    version="1.1"
    viewBox="0 860 420 500"
    width="22px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
      id="arrowLeft"
      transform="scale(1, -1) translate(0, -1500)"
    />
  </svg>
</button>
`;

exports[`ArrowButton DOM rendering renders correctly right variant 1`] = `
<button
  className="right"
  disabled={false}
  onClick={[Function]}
>
  <svg
    className="icon"
    height="36px"
    version="1.1"
    viewBox="0 860 420 500"
    width="22px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
      id="arrowLeft"
      transform="scale(1, -1) translate(0, -1500)"
    />
  </svg>
</button>
`;

exports[`ArrowButton DOM rendering renders correctly when disabled 1`] = `
<button
  className="right"
  disabled={true}
  onClick={[Function]}
>
  <svg
    className="icon"
    height="36px"
    version="1.1"
    viewBox="0 860 420 500"
    width="22px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
      id="arrowLeft"
      transform="scale(1, -1) translate(0, -1500)"
    />
  </svg>
</button>
`;
